### ISSUE TITLE
Buffer appears unlisted if I do `:tab sb` or `<c-w>T`

### ISSUE THREAD

**Issue by ajitid** [tiagovla/scope.nvim#2]
Steps to reproduce:
- Try to move/copy over a buffer to a new tab using `:tab sb` or `<c-w>T`
- Do `:ls`. You won't see the buffer as it is unlisted
- Do `:setlocal buflisted` followed by `:ls`. You would now see the buffer appearing in the buffer list.

**Comment by tiagovla** [comment]
I think I tried to call ``vim.api.nvim_buf_set_option(0, "buflisted", true)`` on ``TabEnter`` but it was not quite working. I will check it again.

**Comment by tiagovla** [comment]
@ajitid could you check if this fixed it?

**Comment by ajitid** [comment]
Seems like this fixed it, but this also created a new problem. To reproduce:

- Open a file
- Do `:tabnew`
- Do `ls`
- Even though this is a new tab you would see the buffer in previous tab as your alternate buffer here

**Comment by tiagovla** [comment]
I noticed it. I think the problem is that the TabEvents happen before `BufEnter`. I'm trying to fix this without introducing custom commands.

**Comment by tiagovla** [comment]
I think it now works with `:tabnew` and `<c-w>T`. It still doesn't work with `:tab sb`, I think it's an upstream issue, it doesn't trigger the `TabNewEntered` event. They fixed `<c-w>T` https://github.com/neovim/neovim/issues/4979 a while ago.

**Comment by ajitid** [comment]
Yep, I've subscribed to that issue. Let's keep this open until upstream is resolved.

**Comment by ajitid** [comment]
> I think it now works with `:tabnew` and `<c-w>T`. It still doesn't work with `:tab sb`, I think it's an upstream issue, it doesn't trigger the `TabNewEntered` event. They fixed `<c-w>T` [neovim/neovim#4979](https://github.com/neovim/neovim/issues/4979) a while ago.
> 
> EDIT: here's the issue [neovim/neovim#4334](https://github.com/neovim/neovim/issues/4334)

The neovim issue you mentioned has been fixed. Though I haven't checked if it fixes this ticket as well.

**Comment by ajitid** [comment]
> Seems like this fixed it, but this also created a new problem. To reproduce:
> 
> * Open a file
> * Do `:tabnew`
> * Do `:ls`
> * Even though this is a new tab you would see the buffer in your previous tab as your alternate buffer here

Just checked, this isn't fixed.
